feat: text editor with slate

berdal84 opened the pull request:
## Editor
### Editing a note
![image](https://github.com/notes-org/notes-frontend/assets/942052/103a95b3-4e31-4618-abd0-21a52125ef0c)

## ReadOnlyEditor (same note in read-only mode)
![image](https://github.com/notes-org/notes-frontend/assets/942052/341b6c1e-462a-45d7-8f1e-2306bf9967f4)

## Evolutions

To update the editor in the future, it looks very simple. There is two ways:
1. Rendering: edit our `useRenderer.tsx` you can customize the rendering (tsx)
2. Behavior: we can implement a `withXorYFeature( editor )` method to inject functionalities to the Slate editor (there is a history plugin, and many others to inspire from).

My opinion is we should merge this branch if the review goes well.

rjmello commented on src/components/editor/Editor.tsx:
```diff
@@ -61,40 +95,168 @@ function Editor( { resource }: EditorProps ) {
+                <BlockButton format="right" icon="format_align_right" />
+                <BlockButton format="justify" icon="format_align_justify" />
+              </Box>
+              <Editable
```
Can we set the default height of the text box to 150-200px or so?

berdal84 replied:
Yep, will change that.

rjmello commented on src/utils/ApiClient.ts:
```diff
@@ -249,7 +249,10 @@ export namespace ApiClient {
         export function createNote(note: api.NotePOST, resource: api.Resource): Promise<AxiosResponse<api.Note>> {
-            return httpClient.post(API_PATH.NOTES, note, { params: { url: resource.url } });
+            const _note = {
+                content: JSON.stringify(note.content) // TODO: allow an object instead of string on the backend API? Would be symetric
```
JSON is transmitted as a string over HTTP. Thus, we will always need to call `JSON.stringify()`, or a similar function, before sending it to the API.

berdal84 replied:
Without a stringify the API rejects the body, a string is expected, not an JSON object.

berdal84 replied:
> JSON is transmitted as a string over HTTP. Thus, we will always need to call `JSON.stringify()`, or a similar function, before sending it to the API.

The http client (axios) does it for us, when we pass a stringified object the client may escape special characters and a simple string is sent.

rjmello replied:
Yeah, I'm only pointing out that there is nothing to change on the back-end API, as the comment suggested, because JSON must always be converted to a string before it can be transmitted over HTTP. I don't have any preference on whether we rely on `axios` or `JSON.stringify()` for this.

rjmello replied:
Ah, I see your point. The API expects a valid nested JSON object, represented as a string, but will reject strings that are not valid JSON. I'll add some logic on the back-end so we don't have to perform this extra serialization step.

berdal84 commented:
Just fixed the height on the note editor. I also added a border since it was looking weird with all this empty space.
![image](https://github.com/notes-org/notes-frontend/assets/942052/780963d7-394e-4567-8f29-5ab5847f34ea)
